src/V3Gate.cpp

```diff
@@ -524,6 +524,54 @@ class GateClkDecomp final {
     static void apply(GateGraph& graph) { GateClkDecomp{graph}; }
 };
 
+//######################################################################
+// Iterate over vertices looking for wide expressions to exclude from inline
+
+class GateWideExprExclude final {
+    size_t m_statExcluded = 0;  // Statistic tracking
+
+    void exclude(GateVarVertex* const vVtxp) {
+        if (!vVtxp->varScp()->isWide()) return;
+        if (vVtxp->inEmpty() || !vVtxp->inEdges().frontp() || !vVtxp->inEdges().frontp()->fromp())
+            return;  // No driver, no exclusion.
+
+        const GateLogicVertex* const lVtxp
+            = vVtxp->inEdges().frontp()->fromp()->as<GateLogicVertex>();
+
+        if (const AstActive* const primaryActivep = lVtxp->activep()) {
+            // Exclude from inlining variables READ multiple times that are initialized by wide
+            // assigns.
+            int reads = 0;
+            for (const V3GraphEdge* const edgep : vVtxp->outEdges().unlinkable()) {
+                if (!edgep || !edgep->top()) continue;
+                if (const GateLogicVertex* const lvp = edgep->top()->as<GateLogicVertex>()) {
+                    // To decouple actives thus simplifying scheduling, exclude only those
+                    // VarRefs that are referenced under the same active as they were assigned.
+                    if (lvp->activep() == primaryActivep) reads += edgep->weight();
+                }
+            }
+            if (reads > 1 && VN_IS(lVtxp->nodep(), NodeAssign)) {
```

Comment:
When you are checking multiple conditions, check the cheapest, most likely one first, so you don't have to do the expensive ones if the cheap checks can decide. Here you check `VN_IS(lVtxp->nodep(), NodeAssign)` after the loop, event though it's a much simpler condition.